unit/src/ExpressionBuilderTest.C

```diff
@@ -34,45 +35,95 @@ void ExpressionBuilderTest::test()
+    CPPUNIT_ASSERT( std::string(H((z,y,x))) == "z*y*x" );
+  }
 
   // Test single bracket syntax
-  EBTerm z("z"), u("u"), v("v"), w("w");
-  G(x,y,z) = x*y*z;
-  H(u,v,w,x,y,z) = u+v+w+x+y+z;
+  {
+    G(x,y,z) = x * y * z;
+    H(u,v,w,x,y,z) = u + v + w + x + y + z;
 
-  CPPUNIT_ASSERT( std::string(G(a,b,c)) == "1*4*8.9" );
-  CPPUNIT_ASSERT( std::string(H(a,b,c,a,b,c)) == "1+4+8.9+1+4+8.9" );
-  CPPUNIT_ASSERT( std::string(G(a,b,c)+H(a,b,c,a,b,c)) == "1*4*8.9+1+4+8.9+1+4+8.9" );
+    CPPUNIT_ASSERT( std::string(G(a,b,c)) == "1*4*8.9" );
+    CPPUNIT_ASSERT( std::string(H(a,b,c,a,b,c)) == "1+4+8.9+1+4+8.9" );
+    CPPUNIT_ASSERT( std::string(G(a,b,c)+H(a,b,c,a,b,c)) == "1*4*8.9+1+4+8.9+1+4+8.9" );
+
+    H(u,v,w,x,y,z) = u + v + w + G(y,x,x);
+    CPPUNIT_ASSERT( std::string(H(a,b,c,x,y,z)) == "1+4+8.9+y*x*x" );
+  }
 
   // Test associativity
-  EBTerm def   = x - y - z;
-  EBTerm left  = (x - y) - z;
-  EBTerm right = x - (y - z);
+  {
+    EBTerm def   = x - y - z;
+    EBTerm left  = (x - y) - z;
+    EBTerm right = x - (y - z);
 
-  CPPUNIT_ASSERT( std::string(def) == "x-y-z" );
-  CPPUNIT_ASSERT( std::string(left) == "x-y-z" );
-  CPPUNIT_ASSERT( std::string(right) == "x-(y-z)" );
+    CPPUNIT_ASSERT( std::string(def) == "x-y-z" );
+    CPPUNIT_ASSERT( std::string(left) == "x-y-z" );
+    CPPUNIT_ASSERT( std::string(right) == "x-(y-z)" );
+  }
 
   // test comparison operators
-  EBTerm comp1 = (x < y) + (x > y);
-  EBTerm comp2 = (x <= y) + (x >= y);
-  EBTerm comp3 = (x == y) + (x != y);
-
-  CPPUNIT_ASSERT( std::string(comp1) == "(x<y)+(x>y)" );
-  CPPUNIT_ASSERT( std::string(comp2) == "(x<=y)+(x>=y)" );
-  CPPUNIT_ASSERT( std::string(comp3) == "(x=y)+(x!=y)" );
+  {
+    EBTerm comp1 = (x < y) + (x > y);
+    EBTerm comp2 = (x <= y) + (x >= y);
+    EBTerm comp3 = (x == y) + (x != y);
+
+    CPPUNIT_ASSERT( std::string(comp1) == "(x<y)+(x>y)" );
+    CPPUNIT_ASSERT( std::string(comp2) == "(x<=y)+(x>=y)" );
+    CPPUNIT_ASSERT( std::string(comp3) == "(x=y)+(x!=y)" );
+  }
+
+  // test binary functions
+  {
+    EBTerm comp4 = atan2(x , y);
+    CPPUNIT_ASSERT( std::string(comp4) == "atan2(x,y)" );
+  }
 
   // test ifexpr
-  EBTerm if1 = conditional(x < 2 * y, x*x + y, y*y +x);
-  CPPUNIT_ASSERT( std::string(if1) == "if" "(x<2*y,x*x+y,y*y+x)" );
+  {
+    EBTerm if1 = conditional(x < 2 * y, x*x + y, y*y +x);
+    CPPUNIT_ASSERT( std::string(if1) == "if" "(x<2*y,x*x+y,y*y+x)" );
+  }
+
+  // test temp id node stringify
+  {
+    EBTerm temp1;
+    EBTerm temp2;
+    CPPUNIT_ASSERT( std::string(temp1) != std::string(temp2) );
+  }
+
+  // test substitution
+  {
+    // plog substitution
+    EBTerm u = log(x/a);
+    u.substitute(EBLogPlogSubstitution(b));
+    CPPUNIT_ASSERT( std::string(u) == "plog(x/1,4)" );
+
+    // single substitution in a ternary
+    EBTerm v = conditional(x < y, x * y, x / y);
+    v.substitute(EBTermSubstitution(x, a));
+    CPPUNIT_ASSERT( std::string(v) == "if(1<y,1*y,1/y)" );
```

Comment:
Oh right, I've had that before...

On Fri, Feb 26, 2016 at 3:16 PM Cody Permann <notifications@github.com>
wrote:

> In unit/src/ExpressionBuilderTest.C
> <https://github.com/idaholab/moose/pull/6449#discussion_r54309880>:
>
> > +    EBTerm temp1;
> > +    EBTerm temp2;
> > +    CPPUNIT_ASSERT( std::string(temp1) != std::string(temp2) );
> > +  }
> > +
> > +  // test substitution
> > +  {
> > +    // plog substitution
> > +    EBTerm u = log(x/a);
> > +    u.substitute(EBLogPlogSubstitution(b));
> > +    CPPUNIT_ASSERT( std::string(u) == "plog(x/1,4)" );
> > +
> > +    // single substitution in a ternary
> > +    EBTerm v = conditional(x < y, x * y, x / y);
> > +    v.substitute(EBTermSubstitution(x, a));
> > +    CPPUNIT_ASSERT( std::string(v) == "if(1<y,1*y,1/y)" );
>
> This is probably the reason for your failed pre-check.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6449/files#r54309880>.
>